### ISSUE TITLE
Allow all directives in line comments

### ISSUE THREAD

**Issue by mdjermanovic** [eslint/eslint#14575]
This is an issue to track the development of https://github.com/eslint/rfcs/pull/34.

* [RFC Document](https://github.com/eslint/rfcs/tree/main/designs/2019-allow-all-directives-in-line-comments)

**Comment by aladdin-add** [comment]
I made a simple implementation, and found ~41 inline comments which were not intended as a directive, but parse as directives.

```bash
➜  eslint git:(issue14575) npm run lint

> eslint@7.27.0 lint
> node Makefile.js lint

Validating JavaScript files

/Users/weiran/repo/github/eslint/lib/rules/newline-before-return.js
  137:51  error  'return' is defined but never used     no-unused-vars
  137:58  error  'at' is defined but never used         no-unused-vars
  137:61  error  'beginning' is defined but never used  no-unused-vars
  137:71  error  'of' is defined but never used         no-unused-vars
  137:74  error  'script' is defined but never used     no-unused-vars

/Users/weiran/repo/github/eslint/tests/lib/rules/id-blacklist.js
  1013:20  error  'declared' is defined but never used       no-unused-vars
  1013:29  error  'in' is defined but never used             no-unused-vars
  1013:32  error  'the' is defined but never used            no-unused-vars
  1013:36  error  'given' is defined but never used          no-unused-vars
  1013:42  error  'source' is defined but never used         no-unused-vars
  1013:49  error  'code' is defined but never used           no-unused-vars
  1013:54  error  'are' is defined but never used            no-unused-vars
  1013:58  error  'not' is defined but never used            no-unused-vars
  1013:62  error  'excluded' is defined but never used       no-unused-vars
  1013:71  error  'from' is defined but never used           no-unused-vars
  1013:76  error  'consideration' is defined but never used  no-unused-vars

/Users/weiran/repo/github/eslint/tests/lib/rules/id-denylist.js
  1013:20  error  'declared' is defined but never used       no-unused-vars
  1013:29  error  'in' is defined but never used             no-unused-vars
  1013:32  error  'the' is defined but never used            no-unused-vars
  1013:36  error  'given' is defined but never used          no-unused-vars
  1013:42  error  'source' is defined but never used         no-unused-vars
  1013:49  error  'code' is defined but never used           no-unused-vars
  1013:54  error  'are' is defined but never used            no-unused-vars
  1013:58  error  'not' is defined but never used            no-unused-vars
  1013:62  error  'excluded' is defined but never used       no-unused-vars
  1013:71  error  'from' is defined but never used           no-unused-vars
  1013:76  error  'consideration' is defined but never used  no-unused-vars

/Users/weiran/repo/github/eslint/tests/lib/rules/no-promise-executor-return.js
   89:19  error  'Promise' is already defined as a built-in global variable  no-redeclare
   89:19  error  'Promise' is defined but never used                         no-unused-vars
   89:27  error  'doesn't' is defined but never used                         no-unused-vars
   89:35  error  'exist' is defined but never used                           no-unused-vars
  100:19  error  'Promise' is already defined as a built-in global variable  no-redeclare
  100:27  error  'is' is defined but never used                              no-unused-vars
  100:30  error  'shadowed' is defined but never used                        no-unused-vars

/Users/weiran/repo/github/eslint/tests/lib/rules/no-setter-return.js
  227:19  error  'object' is defined but never used    no-unused-vars
  227:26  error  'doesn't' is defined but never used   no-unused-vars
  227:34  error  'exist' is defined but never used     no-unused-vars
  235:19  error  'object' is already defined           no-redeclare
  235:26  error  'is' is defined but never used        no-unused-vars
  235:29  error  'shadowed' is defined but never used  no-unused-vars

/Users/weiran/repo/github/eslint/tests/lib/rules/require-await.js
  46:19  error  'await' is defined but never used  no-unused-vars

✖ 41 problems (41 errors, 0 warnings)
```

**Comment by mdjermanovic** [comment]
> I made a simple implementation, and found, in eslint codebase, ~41 inline comments which were not intended as a directive, but parse as directives.

I noticed the same, but it's actually 8 comments causing 41 errors (each comment defines multiple variables).

> I do know the rfc has been accepted, but is there a possibility to reconsider the issue?( as [the concern](https://github.com/eslint/rfcs/pull/34#issuecomment-820364901) in the rfc is not fully addressed.)

It was addressed in [Backwards Compatibility Analysis](https://github.com/captbaritone/rfcs/blob/466125076c80756a6a81c3efaee19cc51707f785/designs/2019-allow-all-directives-in-line-comments/README.md#backwards-compatibility-analysis), by treating this as a breaking change.

**Comment by mdjermanovic** [comment]
@aladdin-add are you already working on this? I was about to start writing code, but if you've already started you can take it.

**Comment by aladdin-add** [comment]
fine, I will take it, will make a PR in 1~2 days. thanks! ❤️

**Comment by aladdin-add** [comment]
oops.. hit the wrong button. sorry!

**Comment by aladdin-add** [comment]
reopened, as it has been reverted in 41617ec3c4bc8bd1ba5f66521185be1566e6f5f4.

**Comment by glen-84** [comment]
Could it be added for `eslint-disable` and `eslint-enable`, at least?

I would pair this with the deprecation and later removal of `global`, `export`, and `eslint`, which are far too generic for directives.

**Comment by mdjermanovic** [comment]
Related discussion: https://github.com/eslint/eslint/issues/14960

**Comment by nzakas** [comment]
**TSC Summary:** This accepted proposal is to allow all inline directives in line comments in addition to block comments. When we tried to implement this, the result was fairly disruptive to the ecosystem. This issue has say open for over a year without any progress.

**TSC Question:** Should we close that as "won't fix" now that we know the result when we tried to do this last?

**Comment by glen-84** [comment]
I hope that the TSC will consider my suggestion of prefixing comment directives, in order to avoid ambiguity.

`exported` -> `eslint-exported`
`globals` -> `eslint-globals`
`global` -> `eslint-global`
`eslint` -> `eslint-rules`

(or a syntax even less likely to be part of a non-directive comment, like `eslint::exported`)

This would be consistent with other directives like `eslint-enable`, and avoid [the issue](https://github.com/eslint/eslint/issues/14960) that forced the initial revert.

**Comment by sam3k** [comment]
Per the decision made in the [2024-08-24](https://github.com/eslint/tsc-meetings/pull/451) tsc-meeting, we are closing this issue and won't be moving forward with it.
